examples/compose/plot_column_transformer.py

```diff
@@ -39,95 +33,130 @@
+                                      return_X_y=True)
+X_test, y_test = fetch_20newsgroups(random_state=1,
+                                    subset='test',
+                                    categories=categories,
+                                    remove=('footers', 'quotes'),
+                                    return_X_y=True)
 
-    def fit(self, x, y=None):
-        return self
+##############################################################################
+# Each feature comprises meta information about that post, such as the subject
+# and the body of the news post.
 
-    def transform(self, posts):
-        return [{'length': len(text),
-                 'num_sentences': text.count('.')}
-                for text in posts]
+print(X_train[0])
 
+##############################################################################
+# Creating transformers
+# ---------------------
+#
+# First, we would like a transformer that extracts the subject and
+# body of each post. Since this is a stateless transformation (does not
+# require state information from training data), we can define a function that
+# performs the data transformation then use
+# :class:`~sklearn.preprocessing.FunctionTransformer` to create a scikit-learn
+# transformer.
 
-class SubjectBodyExtractor(TransformerMixin, BaseEstimator):
-    """Extract the subject & body from a usenet post in a single pass.
 
-    Takes a sequence of strings and produces a dict of sequences.  Keys are
-    `subject` and `body`.
-    """
-    def fit(self, x, y=None):
-        return self
+def subject_body_extractor(posts):
+    # construct object dtype array with two columns
+    # first column = 'subject' and second column = 'body'
+    features = np.empty(shape=(len(posts), 2), dtype=object)
+    for i, text in enumerate(posts):
+        # temporary variable `_` stores '\n\n'
+        headers, _, body = text.partition('\n\n')
+        # store body text in second column
+        features[i, 1] = body
 
-    def transform(self, posts):
-        # construct object dtype array with two columns
-        # first column = 'subject' and second column = 'body'
-        features = np.empty(shape=(len(posts), 2), dtype=object)
-        for i, text in enumerate(posts):
-            headers, _, bod = text.partition('\n\n')
-            features[i, 1] = bod
+        prefix = 'Subject:'
+        sub = ''
+        # save text after 'Subject:' in first column
+        for line in headers.split('\n'):
+            if line.startswith(prefix):
+                sub = line[len(prefix):]
+                break
+        features[i, 0] = sub
 
-            prefix = 'Subject:'
-            sub = ''
-            for line in headers.split('\n'):
-                if line.startswith(prefix):
-                    sub = line[len(prefix):]
-                    break
-            features[i, 0] = sub
+    return features
 
-        return features
 
+SubjectBodyExtractor = FunctionTransformer(subject_body_extractor)
```

Comment:
Since this is an instance:

```suggestion
subject_body_extractor = FunctionTransformer(subject_body_extractor)
```

Reply:
`subject_body_transformer` so it's not the same name as the function? Not that this matters, though if we make it the same name we should also do this for `text_stats_transformer` to be consistent.